nova/tests/unit/compute/test_api.py

```diff
@@ -2230,6 +2235,29 @@ def test_resize_same_host_and_allowed(self):
     def test_resize_same_host_and_not_allowed(self):
         self._test_resize(same_host=True, allow_same_host=False)
 
+    def test_resize_with_volume_in_valid_state(self):
+        # we pass 2 volumes, so we can test both valid states without calling
+        # _test_resize twice
+        bdms = objects.BlockDeviceMappingList(objects=[
+                objects.BlockDeviceMapping(volume_id=uuids.volume_id),
+                objects.BlockDeviceMapping(volume_id=uuids.volume_id)])
+        fake_volume1 = {'status': "in-use", 'attach_status': 'attached'}
+        fake_volume2 = {'status': "available", 'attach_status': 'detached'}
+        # side_effect returns the values in the iterable on consecutive calls
+        with mock.patch.object(self.compute_api.volume_api, 'get',
+                               side_effect=(fake_volume1, fake_volume2)):
+            self._test_resize(bdms=bdms)
```

Comment:
We could check that `self.compute_api.volume_api.get()` got called 2 times. If the tested code would just make one call, you wouldn't test both volume cases you wanted to test.